qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java

```diff
@@ -186,15 +199,79 @@ private static void doCodeActionToDisableValidation(Template template, List<Diag
 		codeActions.add(disableValidationForTemplateQuickFix);
 	}
 
+	/**
+	 * Create CodeAction for deprecated `with` Qute syntax.
+	 *
+	 * @param template    the Qute template.
+	 * @param diagnostic  the diagnostic list that this CodeAction will fix.
+	 * @param codeActions the list of CodeActions to perform.
+	 * @throws BadLocationException
+	 *
+	 */
+	private void doCodeActionsForNotRecommendedWithSection(Template template, Diagnostic diagnostic,
+			List<CodeAction> codeActions) {
+		Range withSectionRange = diagnostic.getRange();
+		try {
+			int withSectionStart = template.offsetAt(withSectionRange.getStart());
+			WithSection withSection = (WithSection) template.findNodeAt(withSectionStart + 1);
+			String javaResourceClassName = withSection.getObjectParameter().getName();
+			List<Node> withExpressions = findExpressions(withSection);
+			List<String> letExpressionParameterList = new ArrayList<String>();
+			for (Node expression : withExpressions) {
+				letExpressionParameterList.add(MessageFormat.format("{1}={0}.{1}", javaResourceClassName,
+						((Expression) expression).getContent()));

```

Comment:
Expression#getContent() returns the content of the expression (ex : `name.toString()` from `{name.toString()}`. More you get only expression, you must take care of too parameter (ex: `{#let foo=name}`.

In another words you need to catch object part from expression or parameter. To fix that, I suggest that you replace your 

`List<Node> withExpressions = findExpressions(withSection);`

with:

`List<String> objectPartNames = new ArrayList<>();
findObjectPart(withSection, objectPartNames);
`

and when you loop for children node, you get object part and collect the name in the given objectPartNames.

Please not set the array list objectPartNames in parameter method, give the capability to avoid creating an array list instance on each loop like https://github.com/redhat-developer/quarkus-ls/pull/563/files#diff-40c1df0b4206ee8c5be37ae62a3d2752b4eb564f080cd67fc2bfdb57728b6081R255